static_report/src/components/shared/ColumnCard/ColumnCardHeader.tsx

```diff
@@ -0,0 +1,72 @@
+import { ColorProps, Flex, Icon, Text, Tooltip } from '@chakra-ui/react';
+import { InfoIcon } from '@chakra-ui/icons';
+import { AiOutlineFileText } from 'react-icons/ai';
+import { BiText, BiQuestionMark } from 'react-icons/bi';
+import { BsCalendarDate } from 'react-icons/bs';
+import { TbCircleHalf, TbCircles } from 'react-icons/tb';
+import { TiSortNumerically } from 'react-icons/ti';
+import { ColumnSchema } from '../../../sdlc/single-report-schema';
+import { formatTruncateString } from '../../../utils/formatters';
+import { checkColumnCategorical } from '../../../utils/transformers';
+import { SRTooltip } from '../../SingleReport/SRTooltip';
+
+type Props = { columnDatum: ColumnSchema };
+export function ColumnCardHeader({ columnDatum }: Props) {
+  const { description, name, schema_type } = columnDatum;
+  const { backgroundColor, icon } = getIconForColumnType(columnDatum);
+  return (
+    <Flex
+      p={2}
+      bg={'white'}
+      justify={'space-between'}
+      alignItems={'center'}
+      width={'100%'}
+      borderTopRadius={'inherit'}
+    >
+      <Flex alignItems={'center'}>
+        <Tooltip label={schema_type} shouldWrapChildren={true}>
+          <Icon
+            mt={1}
+            mx={2}
+            p={1}
+            rounded={'md'}
+            color={'white'}
+            backgroundColor={backgroundColor}
+            as={icon}
+            boxSize={7}
+          />
+        </Tooltip>
+        <Text fontWeight={'semibold'} fontSize={'3xl'}>
+          {formatTruncateString(name, 10)}
+        </Text>
+      </Flex>
+      <SRTooltip label={description} prefix={' - via '}>
+        <InfoIcon color="gray.400" boxSize={'20px'} mr={3} />
+      </SRTooltip>
+    </Flex>
+  );
+}
+
+function getIconForColumnType(columnDatum: ColumnSchema): {
+  backgroundColor: ColorProps['color'];
+  icon: any; //IconType not provided
+} {
+  const { type } = columnDatum;
+  const isCategorical = checkColumnCategorical(columnDatum);
+
+  if (isCategorical)
```

Comment:
I feel including brackets in `if` statement is more readable.

Maybe you can use the `switch` statement here.

Reply:
I'm not a big fan of case/switch for the reason that conditional expressions are needed, and make it way harder to read. e.g. `switch(true) { ... }` 🤮 

See: https://stackoverflow.com/questions/2696436/switch-case-with-conditions
 
opting for explicit `{ .. }` instead